internal/access/credential.go

```diff
@@ -56,7 +62,14 @@ func UpdateCredential(c *gin.Context, user *models.Identity, newPassword string)
 		return err
 	}
 
-	return updateCredential(c, user, newPassword, isSelf)
+	if err := updateCredential(c, user, newPassword, isSelf); err != nil {
+		return err
+	}
+
+	// if the request is from an admin, the infra user may not exist yet, so create the
+	// provider_user if it's missing.
+	_, _ = data.CreateProviderUser(rCtx.DBTxn, data.InfraProvider(rCtx.DBTxn), user)
```

Comment:
We ignored the error here previously. It's probably ok, but I wonder if we should handle it for consistency. The `data.CreateProviderUser` function already returns no error if the entry already exists.